inf-clojure.el

```diff
@@ -770,6 +824,61 @@ Return the number of nested sexp the point was over or after."
   (setq-local eldoc-documentation-function #'inf-clojure-eldoc)
   (apply #'eldoc-add-command inf-clojure-extra-eldoc-commands))
 
-(provide 'inf-clojure)
+;;;; Lumo
+;;;; ====
+
+(defgroup lumo nil
+  "Run an external Lumo process (REPL) in an Emacs buffer."
+  :group 'inf-clojure)
 
+(defcustom inf-clojure-lumo-command "lumo"
+  "The command used to launch lumo."
+  :type 'string
+  :group 'lumo)
+
+(defcustom inf-clojure-lumo-args "-d"
+  "The command arguments used to launch lumo."
+  :type 'string
+  :group 'lumo)
+
+(defcustom inf-clojure-lumo-hostname "localhost"
+  "The hostname used to launch lumo.
+Note that only if port is specified this will be used."
+  :type 'string
+  :group 'lumo)
+
+(defcustom inf-clojure-lumo-port ""
+  "The port used to launch lumo.
+Note that if port is specified, inf-clojure will connect to the
+REPL using HTTP socket.  No process will be actually started."
+  :type '(choice string integer)
+  :group 'lumo)
+
+;; AR - TODO Alternatively you can specify a command string that will be called,
+;; which should return a string.
+(defcustom inf-clojure-lumo-classpath-generator "cp"
```

Comment:
This approach could be extended to all REPL flavors, the all need a classpath to be built if using the "command" approach. Socket of course will already have everything setup. Lumo does not differ in this aspect so we could even get rid of all this. I personally settled on the socket connection but it is handy sometimes to start things in place.